test/tests/providers/global/loki.all.rules.test.yml

```diff
@@ -0,0 +1,99 @@
+---
+rule_files:
+  - loki.all.rules.yml
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'cortex_ring_members{app="loki-compactor", cluster_id="zj88t", cluster_type="workload_cluster", container="compactor", customer="giantswarm", installation="gorilla", instance="10.7.116.221:3100", job="zj88t-prometheus/workload-zj88t/0", name="compactor", namespace="loki", node="ip-10-6-2-178.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-compactor-676b8c897b-rq298", provider="aws", service_priority="highest", state="Unhealthy"}'
+        values: "0+0x20 1+0x160"  # 1 unhealthy value after 20 minutes
+      - series: 'loki_panic_total{app="loki-compactor", cluster_id="zj88t", cluster_type="workload_cluster", container="compactor", customer="giantswarm", installation="gorilla", instance="10.7.116.221:3100", job="zj88t-prometheus/workload-zj88t/0", namespace="loki", node="ip-10-6-2-178.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-compactor-676b8c897b-rq298", provider="aws", service_priority="highest"}'
+        values: "0+0x20 1+0x160"  # 1 panic after 20 minutes
+      - series: 'loki_request_duration_seconds_count{app="loki-distributor", cluster_id="zj88t", cluster_type="workload_cluster", container="distributor", customer="giantswarm", installation="gorilla", instance="10.7.75.90:3100", job="zj88t-prometheus/workload-zj88t/0", method="POST", namespace="loki", node="ip-10-6-2-141.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-distributor-74b78f5559-tz6zs", provider="aws", route="loki_api_v1_push", service_priority="highest", status_code="204", ws="false"}'
+        values: "0+60x180"  # 1 request per second OK for 3 hours
+      - series: 'loki_request_duration_seconds_count{app="loki-distributor", cluster_id="zj88t", cluster_type="workload_cluster", container="distributor", customer="giantswarm", installation="gorilla", instance="10.7.75.90:3100", job="zj88t-prometheus/workload-zj88t/0", method="POST", namespace="loki", node="ip-10-6-2-141.eu-central-1.compute.internal", organization="giantswarm-production", pod="loki-distributor-74b78f5559-tz6zs", provider="aws", route="loki_api_v1_push", service_priority="highest", status_code="503", ws="false"}'
+        values: "0+0x20 0+30x160"  # After 20 minutes, we also have 0.5 rq/s failing 
+    alert_rule_test:
+
+      - alertname: LokiRequestPanics
+        eval_time: 15m  # should be OK after 15 minutes
+        exp_alerts:
```

Comment:
an empty `exp_alerts` mean the alert does not fire right ?

Reply:
Right!